cmd/kubeadm/app/phases/certs/renewal/expiration.go

```diff
@@ -0,0 +1,124 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package renewal
+
+import (
+	"crypto/x509"
+	"fmt"
+	"io"
+	"math"
+	"strings"
+	"text/tabwriter"
+	"time"
+)
+
+// RenewalPeriodDays defines the leng of a period before the certificate expiration when renewal should be
+// planned and executed. This period consider that the default duration for kubeadm certificates is 365d,
+// the fact that kubeadm handles automatic certificate renewal during upgrades, and the release frequence of K8s.
+const RenewalPeriodDays = 180
+
+// ExpirationInfoList define a list of ExpirationInfo
+type ExpirationInfoList []*ExpirationInfo
+
+// InRenewalWindow returns the subset of certificates already set for renewal
+func (el ExpirationInfoList) InRenewalWindow() ExpirationInfoList {
+	l := ExpirationInfoList{}
+	for _, e := range el {
+		if e.IsInRenewalWindow() {
+			l = append(l, e)
+		}
+	}
+	return l
+}
+
+// Write creates a table with the list of certificate exipration info
+func (el ExpirationInfoList) Write(out io.Writer) {
+	// Check expiration
+	w := tabwriter.NewWriter(out, 10, 4, 3, ' ', 0)
+	fmt.Fprintln(w, "CERTIFICATE\tEXPIRES\tRESIDUAL VALIDITY\tSET FOR AUTOMATIC RENEWAL")
+	for _, ce := range el {
+		s := fmt.Sprintf("%s\t%s\t%s\t%-8v",
+			ce.Name,
+			ce.ExpirationDate.Format("Jan 02, 2006 15:04 MST"),
+			humanizeDuration(ce.ResidualValidity()),
+			ce.IsInRenewalWindow(),
+		)
+
+		fmt.Fprintln(w, s)
+	}
+	w.Flush()
+}
+
+// ExpirationInfo defines expriation info for a certificate
+type ExpirationInfo struct {
+	// Name of the certificate
+	// For PKI certificates, it is the name defined in the certsphase package, while for certificates
+	// embedded in the kubeConfig files, it is the kubeConfig file name defined in the kubeadm constants package.
+	// If you use the CertificateRenewHandler returned by Certificates func, handler.Name already contains the right value.
+	Name string
+
+	// The certificate expiration date
+	ExpirationDate time.Time
+}
+
+// newExpirationInfo returns a new ExpirationInfo
+func newExpirationInfo(name string, cert *x509.Certificate) *ExpirationInfo {
+	return &ExpirationInfo{
+		Name:           name,
+		ExpirationDate: cert.NotAfter,
+	}
+}
+
+// IsInRenewalWindow returns true if renewal should be planned and executed for the certificate
+func (e *ExpirationInfo) IsInRenewalWindow() bool {
+	return time.Now().Add(time.Hour * 24 * RenewalPeriodDays).After(e.ExpirationDate)
+}
+
+// ResidualValidity returns the certificate residual validity
+func (e *ExpirationInfo) ResidualValidity() time.Duration {
+	return e.ExpirationDate.Sub(time.Now())
+}
+
+func humanizeDuration(duration time.Duration) string {
```

Comment:
Perhaps use [HumanDuration](https://github.com/kubernetes/kubernetes/blob/746404f82a28e55e0b76ffa7e40306fb88eb3317/staging/src/k8s.io/apimachinery/pkg/util/duration/duration.go#L48)?
kubeadm already uses `ShortHumanDuration` from the same package, so nothing new will be included.

Reply:
Good tip! Done